### PR TITLE
[Feat][Ascend] Add Ascend INT8 linear backend

### PR DESCRIPTION
**Pull request by Big2Wheel**
## What this PR does

Adds an Ascend implementation of Comfy Kitchen's `int8_linear` capability.

- dynamically quantizes FP16/BF16 activations with `torch_npu.npu_dynamic_quant`
- executes the linear layer with `torch_npu.npu_quant_matmul`
- supports scalar and per-output-channel weight scales, optional bias, 2D and batched inputs
- supports the existing ConvRot path and `gelu_tanh`/`swiglu` input activations
- validates the call contract before backend selection, so unsupported shapes, dtypes, activations, and ConvRot group sizes fall back through the normal registry path
- only advertises `int8_linear` when the installed torch-npu exposes `npu_quant_matmul`

This avoids the eager `aten::_int_mm` path, which falls back to CPU on the tested Ascend stack.

## Relationship to #151

This is a stacked follow-up to #151 and intentionally builds on its three commits.

#151 introduces the optional `ascend` backend, its registration/availability handling, the shared INT8 quantize/dequantize capabilities, and NPU device discovery. This PR adds the first compute-heavy consumer of that foundation: Ascend-native INT8 linear/ConvRot execution.

Please review and merge #151 first. Until #151 lands, GitHub shows those base commits in this PR as well; after #151 is merged, this branch can be rebased/updated so the remaining diff contains only the INT8-linear-specific commits.

## User-facing change

On supported Ascend NPU environments, Comfy Kitchen can automatically select the `ascend` backend for compatible INT8 linear layers. Other devices and unsupported Ascend calls retain the existing fallback behavior.

## How this was tested

Environment:

- Ascend 910B3, restricted to one device
- Python 3.12
- PyTorch 2.10.0
- torch-npu 2.10.0.post4

Unit/regression tests:

```text
pytest -q tests/test_ascend_backend.py tests/test_backends.py tests/test_constraints.py tests/test_int8.py
110 passed, 68 skipped
```

```text
python -m ruff check comfy_kitchen/backends/ascend/__init__.py tests/test_ascend_backend.py tests/test_backends.py
All checks passed
```

End-to-end validation:

- Z-Image Turbo INT8 ConvRot checkpoint
- 1920x1088, 8 sampling steps, fixed prompt and seed
- 1,360 automatic `ascend` backend selections for `int8_linear` per run
- no `aten::_int_mm` or CPU fallback
- five generated outputs had the same MD5: `428c244502d03e9c7f99431e2d1070f1`
- previous hot-run times: 13.26 s, 13.16 s, and 12.77 s (mean 13.06 s)
- post-rebase fresh-process validation completed in 16.14 s, including model loading/warm-up effects

The eager baseline was also checked: it explicitly fell back from `aten::_int_mm` to CPU and processed only about 10 INT8 linear calls in roughly 55 seconds, so it was stopped rather than presented as a misleading full-workflow comparison.

Representative real-layer operator measurements (`M=8352`):

| Layer | Dequantized BF16 | Ascend INT8 | Latency reduction |
| --- | ---: | ---: | ---: |
| Attention output | 1.088 ms | 0.938 ms | 13.8% |
| QKV projection | 3.139 ms | 2.102 ms | 33.0% |
| FFN w1 | 2.766 ms | 1.929 ms | 30.3% |
| FFN w2 | 2.870 ms | 1.990 ms | 30.7% |

Numerical comparison against a dequantized reference produced cosine similarity around 0.999958 and relative MAE around 0.89%. These benchmark figures document the tested setup and are not intended as a universal performance guarantee.

### CODE REVIEW

File: tests/test_ascend_backend.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    selected = registry.get_capable_backend(
+        "int8_linear",
+        {
+            "x": x,
+            "weight": torch.ones(16, 32, device=ascend_device, dtype=torch.int8),
+            "weight_scale": torch.ones(16, device=ascend_device),
+            "out_dtype": torch.bfloat16,
+        },
+    )
+    assert selected == "ascend"
```
[the remark this answers is not in the document]
Comment (Big2Wheel) [reply tests/test_ascend_backend.py:123]: Fixed in f58c412. The automatic-selection test now verifies that a forced Ascend int8_linear lookup is declined with BackendNotImplementedError when npu_quant_matmul is absent, and all operator-dependent linear/ConvRot tests use a shared availability marker. Quantize/dequantize tests remain enabled. Verified on Ascend 910B3: 110 passed, 68 skipped; an isolated process with npu_quant_matmul removed also passed the optional-operator simulation.

File: tests/test_backends.py [hunk 2]
```diff
@@ -60,6 +61,16 @@ def test_int8_capabilities_listed(self):
...
 
+        if backends["ascend"]["available"]:
+            ascend_caps = backends["ascend"]["capabilities"]
+            assert ascend_caps == [
+                "dequantize_int8_simple",
+                "dequantize_int8_simple_dtype",
+                "int8_linear",
+                "quantize_int8_rowwise",
+                "quantize_int8_tensorwise",
+            ]
```
[the remark this answers is not in the document]
Comment (Big2Wheel) [reply tests/test_backends.py:73]: Fixed in f58c412. The expected Ascend capability list is now derived from actual torch_npu.npu_quant_matmul availability: int8_linear is included only when the operator exists, while the four quantize/dequantize capabilities remain required. The missing-operator simulation confirmed the resulting list.

**Comment by Big2Wheel** [comment]
Related sibling PR #154 adds Ascend RoPE/RMS-RoPE. Both #153 and #154 branch directly from #151; neither feature PR depends on the other, and #154 contains no INT8-linear changes.
